plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/TrinoNessieCatalog.java

```diff
@@ -237,7 +237,8 @@ public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
         BaseTable table = (BaseTable) loadTable(session, schemaTableName);
         validateTableCanBeDropped(table);
         nessieClient.dropTable(toIdentifier(schemaTableName), true);
-        deleteTableDirectory(fileSystemFactory.create(session), schemaTableName, table.location());
```

Comment:
Initial PR might have copied code from other catalogs and forgot to handle it. 

Now the behaviour is same as Spark integration (expected)